### PR TITLE
[core] Add type hints to public APIs in worker.py

### PR DESCRIPTION
**Pull request by edoakes**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]

File: python/ray/worker.py [hunk 1]
```diff
@@ -577,37 +577,37 @@ def get_dashboard_url():
...
-        _memory=None,
-        _redis_password=ray_constants.REDIS_DEFAULT_PASSWORD,
-        _temp_dir=None,
-        _lru_evict=False,
-        _metrics_export_port=None,
-        _system_config=None,
-        _tracing_startup_hook=None,
+        _enable_object_reconstruction: bool = False,
+        _redis_max_memory: int = None,
+        _plasma_directory: str = None,
```
Comment (simon-mo) [inline python/ray/worker.py:601]:
```suggestion
        _redis_max_memory: Optional[int] = None,
        _plasma_directory: Optional[str] = None,
```

File: python/ray/worker.py [hunk 2]
```diff
@@ -577,37 +577,37 @@ def get_dashboard_url():
...
+        _node_ip_address: str = ray_constants.NODE_DEFAULT_IP,
+        _driver_object_store_memory: int = None,
+        _memory: int = None,
```
Comment (simon-mo) [inline python/ray/worker.py:604]:
```suggestion
        _driver_object_store_memory: Optional[int] = None,
        _memory: Optional[int] = None,
```

File: python/ray/worker.py [hunk 3]
```diff
@@ -577,37 +577,37 @@ def get_dashboard_url():
...
+        _redis_password: str = ray_constants.REDIS_DEFAULT_PASSWORD,
+        _temp_dir: str = None,
```
Comment (simon-mo) [inline python/ray/worker.py:606]:
```suggestion
        _temp_dir: Optional[str] = None,
```

File: python/ray/worker.py [hunk 4]
```diff
@@ -577,37 +577,37 @@ def get_dashboard_url():
...
+        _lru_evict: bool = False,
+        _metrics_export_port: int = None,
+        _system_config: Dict[str, str] = None,
+        _tracing_startup_hook: Callable = None,
```
Comment (simon-mo) [inline python/ray/worker.py:610]:
```suggestion
        _metrics_export_port: Optional[int] = None,
        _system_config: Optional[Dict[str, str]] = None,
        _tracing_startup_hook: Optional[Callable] = None,
```

File: python/ray/worker.py [hunk 5]
```diff
@@ -1554,7 +1554,9 @@ def show_in_dashboard(message, key="", dtype="text"):
 
 @PublicAPI
 @client_mode_hook
-def get(object_refs, *, timeout=None):
+def get(object_refs: Union[ray.ObjectRef, List[ray.ObjectRef]],
+        *,
+        timeout: Optional[float] = None) -> Union[Any, List[Any]]:
```
Comment (simon-mo) [inline python/ray/worker.py:1559]: Alternatively, add generic to it
```
T = Generic("T")
def get(object_refs: Union[ray.ObjectRef[T], List[ray.ObjectRef[T]]],
        *,
        timeout: Optional[float] = None) -> Union[T, List[T]]:

File: python/ray/worker.py [hunk 6]
```diff
@@ -1641,7 +1643,8 @@ def get(object_refs, *, timeout=None):
 
 @PublicAPI
 @client_mode_hook
-def put(value, *, _owner=None):
+def put(value: Any, *,
```
Comment (simon-mo) [inline python/ray/worker.py:1646]: you can make it generic over type isntead of any

**Review by simon-mo — APPROVED** [review]

**Comment by edoakes** [comment]
Thanks @simon-mo will make those changes!

**Pull request merged**